libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationCollectionClient.java

```diff
@@ -68,7 +84,10 @@ public static LocationCollectionClient install(@NonNull Context context, long de
       if (locationCollectionClient == null) {
         locationCollectionClient = new LocationCollectionClient(new LocationEngineControllerImpl(applicationContext,
           LocationEngineProvider.getBestLocationEngine(applicationContext), new SessionIdentifier(defaultInterval)),
-          new HandlerThread("LocationSettingsChangeThread"));
+          new HandlerThread("LocationSettingsChangeThread"),
+          applicationContext.getSharedPreferences(MAPBOX_SHARED_PREFERENCES, Context.MODE_PRIVATE),
+          // Provide empty token as it is not available yet
+          new MapboxTelemetry(applicationContext, "", LOCATION_COLLECTOR_USER_AGENT));
```

Comment:
I am confused here as MapboxTelemetry is the API class we provide for MapSdk and AutoSDK, they will create a instance of MapboxTelemetry to push gesture events etc, but we create another instance here.

Reply:
`MapboxTelemetry` is no longer coupled with location collection and hence becomes responsible for eventing only. You can treat `LocationCollectionClient` as producer of location events, which are delivered to the `api-events` endpoint via `MapboxTelemetry`.